### ISSUE TITLE
Deno Support

### ISSUE THREAD

**Issue by zicklag** [lit/lit#3272]
### Should this be an RFC?

- [X] This is not a substantial change

### Which package is this a feature request for?

SSR (@lit-labs/ssr)

### Description

I'm trying to use Lit.js SSR in the [Astro](https://astro.build) framework with the Deno SSR engine, but Lit.js SSR currently depends on `require()`s of Node.js-only modules which do not work in Deno.

I know SSR is work-in-progress anyway, but it would be great to have a way to use Lit's SSR on a Deno server.

### Alternatives and Workarounds

I may have to try switching to Node.js instead of Deno, but Deno is preferred for my project due to it's ability to deploy to Deno Deploy's hosted service.

**Comment by justinfagnani** [comment]
I think this could actually be large enough the warrant an RFC.

We would have to drastically change our development process to make sure that we don't use anything Deno-incompatible. I'm not even sure what that would look like yet, or how we would find the dependencies we need that work in both.

**Comment by zicklag** [comment]
Yeah, I was on the fence as to whether or not it was substantial enough for an RFC. 

Without looking into the lit source code I couldn't tell if it would be as dramatic as a change to warrant architecture modifications, or simple enough that it would only require some shims for the NodeJS modules that were imported.

---

If this absolutely needs an RFC, feel free to close. I might take a look at this in the future. It depends on where my time ends up, but I'd have to do investigating into the source code before I could prepare a proper RFC.

**Comment by aomarks** [comment]
If the main issue is dependencies on commonjs modules, I would be really curious to see a list of them. Being es modules only would also have the side benefit of allowing ssr to run in the browser, which could be cool for demos etc.

**Comment by zicklag** [comment]
> If the main issue is dependencies on commonjs modules

I think that's the only problem. For me the issue manifests itself when I try to use Vite ( which I think technically just offloads to Rollup ) to bundle `@litdev/ssr` to a single JS file.

It complains that the Node modules can't be bundled, but as far as I can tell, it even handles CommonJS modules fine, it just can't bundle built-in modules like `path`, `process`, etc.

**Comment by kevinpschaaf** [comment]
So it sounds like the problem isn't actually that lit-ssr is depending on commonjs modules, it's that we're depending on built-in node modules? 

We don't do this much fwict, other than `createRequire` in parse5-utils, and transitively via `node-fetch` in the DOM shim (excluding the VM-based entry points, which wouldn't be in scope for Deno usage anyway).

Once we upgrade to the new esm parse5 and can rely on node's built-in `fetch` in the dom-shim, seems like this might fix itself?

**Comment by mhoad** [comment]
I would switch to Deno in a heartbeat if you ended up making those changes. Would simplify the DX by quite a bit.

**Comment by jaydanurwin** [comment]
I'd switch to Deno and Lit myself as well. It's been such a breath of fresh air to work with honestly. Not having to fiddle with NPM and node_modules is nice but mainly being able to just deploy Typescript to the edge and it run at less than 50ms on the edge is the main draw for me.

**Comment by damooo** [comment]
With DSD, it may not be that difficult to support deno-ssr _/\\\_ @justinfagnani
